From 9ae3574797e084c5d01fe9813f19e35a14af897e Mon Sep 17 00:00:00 2001
From: Ren Jianing <jianing.ren@rock-chips.com>
Date: Thu, 24 Dec 2020 09:25:17 +0800
Subject: [PATCH] arm64: dts: rockchip: rk3568-nvr: fix usb host support

We let phy-supply regulator always on to support all usb host port.

For dwc3 host port, which only works at high speed, combphy1_usq
should be deleted at dwc3 node and add 'rockchip,dis-u3otg1-port'
attribute.

Signed-off-by: Ren Jianing <jianing.ren@rock-chips.com>
Change-Id: I3623de2f76ee27ea34fe2a94481319a3ccb7b23b
---
 .../dts/rockchip/rk3568-nvr-demo-v10.dtsi     |  1 +
 arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi  | 28 +++++++++++++------
 2 files changed, 21 insertions(+), 8 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3568-nvr-demo-v10.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-nvr-demo-v10.dtsi
index 80445097cefd..c953374f473f 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-nvr-demo-v10.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-nvr-demo-v10.dtsi
@@ -87,6 +87,7 @@
 &combphy1_usq {
 	pinctrl-names = "default";
 	pinctrl-0 = <&sata_pm_reset>;
+	rockchip,dis-u3otg1-port;
 	status = "okay";
 };
 
diff --git a/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi b/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
index 75fdd4cd5750..b7a899b7789e 100644
--- a/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3568-nvr.dtsi
@@ -108,13 +108,14 @@
 		vin-supply = <&dc_12v>;
 	};
 
-	vcc5v0_otg: vcc5v0-otg-regulator {
+	vcc5v0_host: vcc5v0-host-regulator {
 		compatible = "regulator-fixed";
 		enable-active-high;
 		gpio = <&gpio0 RK_PA5 GPIO_ACTIVE_HIGH>;
 		pinctrl-names = "default";
-		pinctrl-0 = <&vcc5v0_otg_en>;
-		regulator-name = "vcc5v0_otg";
+		pinctrl-0 = <&vcc5v0_host_en>;
+		regulator-name = "vcc5v0_host";
+		regulator-always-on;
 	};
 
 	vcc_1v8: vcc_1v8 {
@@ -378,22 +379,22 @@
 };
 
 &u2phy0_host {
-	phy-supply = <&vcc5v0_otg>;
+	phy-supply = <&vcc5v0_host>;
 	status = "okay";
 };
 
 &u2phy0_otg {
-	vbus-supply = <&vcc5v0_otg>;
+	phy-supply = <&vcc5v0_host>;
 	status = "okay";
 };
 
 &u2phy1_host {
-	phy-supply = <&vcc5v0_otg>;
+	phy-supply = <&vcc5v0_host>;
 	status = "okay";
 };
 
 &u2phy1_otg {
-	phy-supply = <&vcc5v0_otg>;
+	phy-supply = <&vcc5v0_host>;
 	status = "okay";
 };
 
@@ -431,6 +432,17 @@
 	status = "okay";
 };
 
+&usbhost_dwc3 {
+	phys = <&u2phy0_host>;
+	phy-names = "usb2-phy";
+	maximum-speed = "high-speed";
+	status = "okay";
+};
+
+&usbhost30 {
+	status = "okay";
+};
+
 &vdpu {
 	status = "okay";
 };
@@ -459,7 +471,7 @@
 
 &pinctrl {
 	usb {
-		vcc5v0_otg_en: vcc5v0-otg-en {
+		vcc5v0_host_en: vcc5v0-host-en {
 			rockchip,pins = <0 RK_PA5 RK_FUNC_GPIO &pcfg_pull_none>;
 		};
 	};
-- 
2.35.3

